classes/cron_manager.php

```diff
@@ -64,38 +66,32 @@ function() use ($profiler, $timer) {
      * We then check for a new task with the current sample.
      *
      * @param \ExcimerProfiler $profiler
+     * @param float $started
      */
-    public static function on_interval(\ExcimerProfiler $profiler): void {
+    public static function on_interval(\ExcimerProfiler $profiler, float $started): void {
         $log = $profiler->flush();
 
-        $log->rewind();
-        if (!$log->valid()) {
-            debugging('on_interval called with no profile');
-            return;
-        }
-        $sample = $log->current();
-        $log->next();
-        if ($log->valid()) {
-            debugging('extra profile sample found in on_interval');
-        }
+        foreach ($log as $sample) {
```

Comment:
Same thought here regarding performance. For a long running script this could get very heavy. I think its worth storing the count of samples processed, and then chopping off the first N samples that we know are already processed. I think the only additional data to track would be the number of samples, plus the ending task name.

That will reduce this function from n time to constant time, which will keep the footprint low here

Reply:
Calling flush() on ExcimerProfile removes all logs on the profiler, so next time it comes around, it is all new samples. No sample is processed twice.

Reply:
Not sure why we need to track the ending task name.

Reply:
That would only have been to keep the task list state intact if we were picking up in a new loop, Mightve been unnecessary anyway. In the case that its flushed, my comment is already implemented :)